code/obj/item/grenades.dm

```diff
@@ -1458,6 +1460,12 @@ ABSTRACT_TYPE(/obj/item/old_grenade/spawner)
 				name = "hollow pipe frame"
 			src.flags |= NOSPLASH
 
+		if(issnippingtool(W) && state == 3) //pipeshot crafting
```

Comment:
Correct me if I'm wrong, but this appears like it never considers what chem is used as the propellent.

Example:
Pipebombs use the volatility / strength var to determine if a pipebomb will explode or crumble. Your shotgun shells will accept water is a valid propellent and act the same as black powder.

Two suggestions:
1. Re-use the strength check pipebomb code, if the strength is below a certain threshold (e.x they just used water) give them dud rounds.
2. Make the strength of the propellent matter, increase / decrease speed of projectile. Or maybe even give it a damage boost.

Reply:
They now won't craft if you don't use an explosive chem